main.tf

```diff
@@ -48,6 +49,7 @@ module "main" {
       min_count          = node_pool.min_nodes
       name               = key
       preemptible        = node_pool.preemptible
+      service_account    = var.gcp_service_account
```

Comment:
Not sure if this needs to be overriden here. Also not sure what's supposed to happen when this is set to `""` (proposed default value).

Reply:
Regarding the override, I think it is exactly what the change is aiming for: forcing the use of a specific service_account (in CTE CI workflow, the same account we use for every other GCP operation) while creating the node pools.
Regarding the proposed default, I think it should work as before, since [the documentation](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/v23.1.0#node_pools-variable) states that the default value for `service_account` in `node_pools` variable is the empty string.